src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -768,6 +768,54 @@ public InstructionSetInfo(string name, string managedName, InstructionSet instru
             }
         }
 
+        public static IEnumerable<string> AllCpuFamilies()
+        {
+            // Only report these in the help, others are added to CpuFamilyToInstructionSets just for muscle memory after clang/llvm
+            yield return "generic";
+            yield return "sandybridge";
+            yield return "haswell";
+            yield return "ampere";
+            yield return "apple-m1";
+        }
+
+        public static IEnumerable<string> CpuFamilyToInstructionSets(string cpu, TargetArchitecture arch)
+        {
+            string sets = null;
+
+            if (arch == TargetArchitecture.ARM64)
+            {
+                sets = cpu switch
+                {
+                    "generic" => "base,neon",
+
+                    "ampere" => "base,neon,lse",
+
+                    "apple-m1" => "base,neon,aes,crc,dotprod,rdma,sha1,sha2,lse,rcpc",
```

Comment:
We should set this as default for `TargetOS.OSX` here:

https://github.com/dotnet/runtime/blob/a78c46d1dd9ad718505e3b0508e2c72d6fca6b21/src/coreclr/tools/aot/crossgen2/Program.cs#L239

Reply:
(and same in ilc)

Reply:
Done